src/pyuvdata/beam_interface.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) 2022 Radio Astronomy Software Group
+# Licensed under the 2-clause BSD License
+
+"""Definition for BeamInterface object."""
+
+from __future__ import annotations
+
+import copy
+import warnings
+from dataclasses import InitVar, dataclass
+from typing import Literal
+
+import numpy as np
+import numpy.typing as npt
+
+from .analytic_beam import AnalyticBeam
+from .uvbeam import UVBeam
+
+# Other methods we may want to include:
+#  - beam area
+#  - beam squared area
+#  - efield_to_pstokes
+
+
+@dataclass
+class BeamInterface:
+    """
+    Definition for a unified beam interface.
+
+    This object provides a unified interface for UVBeam and AnalyticBeam objects
+    to compute beam response values in any direction.
+
+    Attributes
+    ----------
+    beam : pyuvdata.UVBeam or pyuvdata.AnalyticBeam
+        Beam object to use for computations
+    beam_type : str
+        The beam type, either "efield" or "power".
+    include_cross_pols : bool
+        Option to include the cross polarized beams (e.g. xy and yx or en and ne).
+        Used if beam is a UVBeam and and the input UVBeam is an Efield beam but
+        beam_type is "power".
+        Ignored otherwise (the cross pol inclusion is set by the beam object.)
+
+    """
+
+    beam: AnalyticBeam | UVBeam
+    beam_type: Literal["efield", "power"] | None = None
+    include_cross_pols: InitVar[bool] = True
+
+    def __post_init__(self, include_cross_pols):
+        """
+        Post-initialization validation and conversions.
+
+        Parameters
+        ----------
+        include_cross_pols : bool
+            Option to include the cross polarized beams (e.g. xy and yx or en and ne)
+            for the power beam.
+
+        """
+        if not isinstance(self.beam, UVBeam) and not issubclass(
```

Comment:
why are you using `issubclass(type(self.beam))` instead of simply `isinstance(self.beam, AnalyticBeam)`?

Reply:
Because that doesn't work. It fails for the subclasses because they aren't an instance on an AnalyticBeam, they're an instance of a subclass of AnalyticBeam. This does work if you prefer it: `isinstance(self.beam.__class__, AnalyticBeam)` as does `issubclass(self.beam.__class__, AnalyticBeam)`. I don't care either way.

Reply:
But `isinstance` works on subclasses: 

![image](https://github.com/user-attachments/assets/8cc58226-0a47-4ced-abbb-0e386000f529)